ui/app/components/secret-engine/list.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { dropTask } from 'ember-concurrency';
+import errorMessage from 'vault/utils/error-message';
+
+import type Router from '@ember/routing/router';
+import type Store from '@ember-data/store';
+import type VersionService from 'vault/services/version';
+import type FlashMessageService from 'vault/services/flash-messages';
+import SecretEngineModel from 'vault/models/secret-engine';
+
+/**
+ * @module SecretEngineList handles the display of the list of secret engines, including the filtering.
+ * 
+ * @example
+ * <SecretEngine::List
+    @secretEngineModels={{this.model}}
+    />
+ *
+ * @param {array} secretEngineModels - An array of Secret Engine models returned from query on the parent route.
+ */
+
+interface Args {
+  secretEngineModels: Array<SecretEngineModel>;
+}
+
+export default class SecretListItem extends Component<Args> {
+  @service declare readonly router: Router;
+  @service declare readonly store: Store;
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked secretEngineOptions: Array<string> | [] = [];
+  @tracked selectedEngineType: string | undefined = undefined;
+  @tracked selectedEngineName: string | undefined = undefined;
+  @tracked engineToDisable: string | undefined = undefined;
```

Comment:
I find it a little easier (and better for staying ✨  aligned ✨  on types)  to keep types the same for variables, unless there's an explicit reason we need the value to be `undefined`. So these can just initialize these as empty strings

```suggestion
  @tracked selectedEngineType = '';
  @tracked selectedEngineName = '';
  @tracked engineToDisable = ''
```

Reply:
updated